examples/svm/plot_svm_scale_c.py

```diff
@@ -33,136 +33,144 @@
 
+In the remainder of this example, we will investigate the effect of scaling
+the value of the regularization parameter `C` in regards to the number of
+samples for both L1 and L2 penalty. We will generate some synthetic datasets
+that are appropriate for each type of regularization.
 """
 
 # Author: Andreas Mueller <amueller@ais.uni-bonn.de>
 #         Jaques Grobler <jaques.grobler@inria.fr>
 # License: BSD 3 clause
 
+# %%
+# L1-penalty case
+# ---------------
+# In the L1 case, theory says that prediction consistency (i.e. that under
+# given hypothesis, the estimator learned predicts as well as a model knowing
+# the true distribution) is not possible because of the bias of the L1. It
+# does say, however, that model consistency, in terms of finding the right set
+# of non-zero parameters as well as their signs, can be achieved by scaling
+# `C`.
+#
+# We will now check if by using synthetic data, we can show this effect. Let's
+# first generate some synthetic data. This dataset will be sparse, meaning
+# that only a few features will be informative and useful for the model.
+from sklearn.datasets import make_classification
+
+n_samples, n_features = 100, 300
+X, y = make_classification(
+    n_samples=n_samples, n_features=n_features, n_informative=5, random_state=1
+)
+
+# %%
+# Now, we can define a linear SVC with the `l1` penalty.
+from sklearn.svm import LinearSVC
+
+penalty, loss = "l1", "squared_hinge"
+model_l1 = LinearSVC(penalty=penalty, loss=loss, dual=False, tol=1e-3)
+
+# %%
+# We will compute the mean test score for different values of `C`.
 import numpy as np
+import pandas as pd
+from sklearn.model_selection import validation_curve, ShuffleSplit
+
+Cs = np.logspace(-2.3, -1.3, 10)
+train_sizes = np.linspace(0.3, 0.7, 3)[::-1]
+
+results = {"C": Cs}
+for train_size in train_sizes:
+    cv = ShuffleSplit(train_size=train_size, test_size=0.3, n_splits=50, random_state=1)
+    train_scores, test_scores = validation_curve(
+        model_l1, X, y, param_name="C", param_range=Cs, cv=cv
+    )
+    results[f"fraction: {train_size}"] = test_scores.mean(axis=1)
+results = pd.DataFrame(results).set_index("C")
+
+# %%
+# Finally, we can plot the results with and without scaling the parameter `C`
+# depending of the size of the training set.
 import matplotlib.pyplot as plt
 
-from sklearn.svm import LinearSVC
-from sklearn.model_selection import ShuffleSplit
-from sklearn.model_selection import GridSearchCV
-from sklearn.utils import check_random_state
-from sklearn import datasets
+fig, axes = plt.subplots(nrows=2, sharey=True, figsize=(9, 10))
+
+# plot results without scaling C
+results.plot(ax=axes[0], logx=True)
+axes[0].set_ylabel("CV score")
+axes[0].set_title(f"scaling=No scaling, penalty={penalty!r}, loss={loss!r}")
+
+# plot results by scaling C
+for train_size_idx, fraction in enumerate(results):
+    results_scaled = results[fraction]
+    results_scaled.index *= float(n_samples * train_sizes[train_size_idx])
```

Comment:
I think changing the index here for plotting is a little too subtle. What do you think of:

```python
Cs = np.logspace(-2.3, -1.3, 10)
train_sizes = np.linspace(0.3, 0.7, 3)
labels = [f"fraction: {train_size}" for train_size in train_sizes]

results = {"C": Cs}
for label, train_size in zip(labels, train_sizes):
    cv = ShuffleSplit(train_size=train_size, test_size=0.3, n_splits=50, random_state=1)
    train_scores, test_scores = validation_curve(
        model_l1, X, y, param_name="C", param_range=Cs, cv=cv
    )
    results[label] = test_scores.mean(axis=1)
results = pd.DataFrame(results)

import matplotlib.pyplot as plt

fig, axes = plt.subplots(nrows=1, ncols=2, sharey=True, figsize=(12, 6))

# plot results without scaling C
results.plot(x="C", ax=axes[0], logx=True)
axes[0].set_ylabel("CV score")
axes[0].set_title(f"scaling=No scaling, penalty={penalty!r}, loss={loss!r}")

# plot results by scaling C
for train_size_idx, label in enumerate(labels):
    results_scaled = results[[label]].assign(
        C_scaled=Cs * float(n_samples * train_sizes[train_size_idx])
    )
    results_scaled.plot(x="C_scaled", ax=axes[1], logx=True, label=label)
```

In this case, "C" remains a column in the dataframe.

Reply:
Fine with me. I did the change